src/test/java/uk/gov/pay/products/resources/PaymentResourceTest.java

```diff
@@ -1,32 +1,171 @@
+                product.getExternalId(),
+                product.getDescription(),
+                product.getReturnUrl(),
+                nextUrl);
+        publicApiStub
+                .whenReceiveCreatedPaymentRequestWithBody(expectedPaymentRequestPayload)
+                .respondCreatedWithBody(paymentResponsePayload);
+
+        ValidatableResponse response = givenAuthenticatedSetup()
+                .accept(APPLICATION_JSON)
+                .post(format("/v1/api/products/%s/payments", product.getExternalId()))
+                .then()
+                .statusCode(201);
+
+        List<Map<String, Object>> paymentRecords = databaseHelper.getPaymentsByProductExternalId(product.getExternalId());
+
+        assertThat(paymentRecords.size(), is(1));
+
+        MatcherAssert.assertThat(paymentRecords.get(0), hasKey("id"));
+        MatcherAssert.assertThat(paymentRecords.get(0), hasKey("external_id"));
+        String paymentExternalId = (String) paymentRecords.get(0).get("external_id");
+
+        MatcherAssert.assertThat(paymentRecords.get(0), hasEntry("govuk_payment_id", govukPaymentId));
+        MatcherAssert.assertThat(paymentRecords.get(0), hasKey("product_id"));
+        MatcherAssert.assertThat(paymentRecords.get(0), hasEntry("next_url", nextUrl));
+        MatcherAssert.assertThat(paymentRecords.get(0), hasEntry("status", "SUCCESS"));
+        MatcherAssert.assertThat(paymentRecords.get(0), hasKey("date_created"));
+
+        response
+                .body("external_id", is(paymentExternalId))
+                .body("govuk_payment_id", is(govukPaymentId))
+                .body("next_url", is(nextUrl))
+                .body("product_external_id", is(product.getExternalId()))
+                .body("status", is("SUCCESS"))
+                .body("_links", hasSize(1))
+                .body("_links[0].href", matchesPattern(paymentsUrl + paymentExternalId))
+                .body("_links[0].method", is(HttpMethod.GET))
+                .body("_links[0].rel", is("self"));
+    }
+
+    @Test
+    public void createAPayment_shouldFail_whenNotAuthenticated() throws Exception {
+        givenSetup()
+                .accept(APPLICATION_JSON)
+                .post(format("/v1/api/products/%s/payments", "payment-id"))
+                .then()
+                .statusCode(401);
+    }
+
+    @Test
+    public void createAPayment_shouldFail_whenProductIsNotFound() throws Exception {
+        String unknownProductId = "unknown-product-id";
+
+        givenAuthenticatedSetup()
+                .accept(APPLICATION_JSON)
+                .post(format("/v1/api/products/%s/payments", unknownProductId))
+                .then()
+                .statusCode(404)
+                .body("errors", hasSize(1))
+                .body("errors[0]", is(format("Product with product id %s not found.", unknownProductId)));
+    }
+
+    @Test
+    public void createAPayment_shouldFail_whenDownstreamError() throws Exception {
```

Comment:
Should we report downstream auth errors differently (perhaps an in valid pay_token) than just a 500 🤔 .

Reply:
As discussed nothing more is needed here.